### ISSUE TITLE
Implement page cache for vinyl

### ISSUE THREAD

**Issue by locker** [tarantool/tarantool#3202]
Vinyl only caches tuples and tuple chains. If a tuple is not found in the cache, it has to read run files. The latter incurs severe overhead even if the pages being read happen to be found in the system page cache, because run files are compressed and we have to decompress them on each read. Besides, relying on system page cache eviction algorithm may result in sub-optimal performance. That said, we should probably implement our own page cache which would store decompressed pages.

**Comment by kostja** [comment]
I disagree with the proposed solution for the problem. Before adding page cache let's make sure the page size is variable and higher levels are not compressed. We also need to make sure compression can be disabled for wal and vinyl, for the latter on per-index basis.
